api/models/sub_criteria.py

```diff
@@ -0,0 +1,32 @@
+import inspect
+from dataclasses import dataclass
+from typing import List, Dict
+from api.models.theme import Theme
+from stringcase import lowercase
+from dataclasses import dataclass
+from dataclass_dict_convert import dataclass_dict_convert
+
+
+@dataclass_dict_convert()
```

Comment:
This annotation is pretty cool - looking at [the docs](https://pypi.org/project/dataclass-dict-convert/) it looks like it does it for nested lists, are the `__post_init__` and `from_filtered_dict` still necessary?

Reply:
So the post_init and from_filtered_dict work together to ensure we can create class instances down through every linked subclass (from a dictionary), without the post_init we dont get the nested lists as a dataclass (and this then nobbles the dict_convert :[).